lib/task.ts

```diff
@@ -0,0 +1,321 @@
+import {
+  sparqlEscapeUri,
+  sparqlEscapeString,
+  sparqlEscapeDateTime,
+  uuid,
+} from 'mu';
+import { v4 as uuidv4 } from 'uuid';
+
+import { querySudo as query, updateSudo as update } from '@lblod/mu-auth-sudo';
+import {
+  TASK_TYPE,
+  PREFIXES,
+  STATUS_BUSY,
+  STATUS_FAILED,
+  ERROR_URI_PREFIX,
+  TASK_HARVESTING_OPARL,
+  ERROR_TYPE,
+  MU_SPARQL_ENDPOINT,
+  TASK_URI_PREFIX,
+  CONTAINER_URI_PREFIX,
+  HARVEST_COLLECTION_URI_PREFIX,
+  REMOTE_DATA_OBJECT_URI_PREFIX,
+  JOB_URI_PREFIX,
+  OPARL_TO_ELI_SERVICE_URI,
+  JOB_HARVESTING_OPARL,
+  IMPORT_GRAPH_URI_PREFIX,
+  STATUS_SCHEDULED,
+} from '../constants';
+import { parseResult } from './utils';
+const connectionOptions = {
+  sparqlEndpoint: MU_SPARQL_ENDPOINT,
+  mayRetry: true,
+};
+export async function failBusyImportTasks() {
+  const queryStr = `
+      PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+      PREFIX adms: <http://www.w3.org/ns/adms#>
+      PREFIX dct: <http://purl.org/dc/terms/>
+      PREFIX task: <http://redpencil.data.gift/vocabularies/tasks/>
+      DELETE {
+        GRAPH ?g {
+          ?task adms:status ${sparqlEscapeUri(STATUS_BUSY)} .
+          ?task dct:modified ?modified.
+        }
+      }
+      INSERT {
+        GRAPH ?g {
+         ?task adms:status ${sparqlEscapeUri(STATUS_FAILED)} .
+         ?task dct:modified ${sparqlEscapeDateTime(new Date())}.
+        }
+      }
+      WHERE {
+        GRAPH ?g {
+          ?task a ${sparqlEscapeUri(TASK_TYPE)};
+                adms:status ${sparqlEscapeUri(STATUS_BUSY)};
+                task:operation ?operation.
+          VALUES ?operation {
+            ${sparqlEscapeUri(TASK_HARVESTING_OPARL)}
+          }
+          OPTIONAL { ?task dct:modified ?modified. }
+        }
+      }
+     `;
+  try {
+    await update(queryStr, {}, { mayRetry: true });
+  } catch (e) {
+    console.warn(
+      'WARNING: failed to move busy tasks to failed status on startup.',
+      e,
+    );
+  }
+}
+  
+export async function isTask(subject) {
+  //TODO: move to ask query
```

Comment:
would remove this comment
also would just add a limit 1 to the query instead if this is about efficiency